src/vm/variables.rs

```diff
@@ -50,11 +50,18 @@ pub fn lookup_reserved_variable(
                 Ok(Some(sender))
             }
             NativeVariables::ContractCaller => {
-                let sender = env
+                let caller = env
                     .caller
                     .clone()
-                    .ok_or(RuntimeErrorType::NoSenderInContext)?;
-                Ok(Some(sender))
+                    .ok_or(RuntimeErrorType::NoCallerInContext)?;
+                Ok(Some(caller))
+            }
+            NativeVariables::TxSponsor => {
+                let sponsor = match env.sponsor.clone() {
+                    None => Value::none(),
+                    Some(p) => Value::some(Value::Principal(p))?,
```

Comment:
Nit: Here, the `Value::some(..)` constructor can throw an error in the event that the constructed object would be "too large" (or otherwise violate the Clarity object size limits). However, since we _know_ that a `Some(Principal)` is a valid Clarity object, we should treat such a failure as a panic (i.e., use `expect()`)